apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/CollectorDataModal.vue

```diff
@@ -129,24 +130,29 @@ const handleClickConfirm = async () => {
 };
 
 /* API */
-const apiQueryHelper = new ApiQueryHelper();
-const fetchSecrets = async (provider: string) => {
-    try {
-        apiQueryHelper.setFilters([{ k: 'provider', v: provider, o: '=' }]);
+const apiQueryHelper = new ApiQueryHelper().setCountOnly();
+const fetchSecrets = async (provider: string, serviceAccounts: string[]) => {
+    apiQueryHelper.setFilters([{ k: 'provider', v: provider, o: '=' }]);
 
-        const results = await SpaceConnector.client.secret.secret.list({
+    if (serviceAccounts?.length) {
+        if (state.isExcludeFilter) apiQueryHelper.addFilter({ k: 'service_account_id', v: serviceAccounts, o: '!=' });
+        else apiQueryHelper.addFilter({ k: 'service_account_id', v: serviceAccounts, o: '=' });
+    }
+    try {
+        const { total_count } = await SpaceConnector.client.secret.secret.list({
             query: apiQueryHelper.data,
         });
-        state.secrets = results.results;
+        state.secretsCount = total_count;
     } catch (e) {
         ErrorHandler.handleError(e);
-        state.secrets = [];
+        state.secretsCount = 0;
     }
 };
 
 watch([() => collectorDataModalState.selectedCollector, () => collectorDataModalState.visible], async ([selectedCollector, visible]) => {
     if (!selectedCollector || !visible) return;
-    await fetchSecrets(selectedCollector.provider);
+    await fetchSecrets(selectedCollector.provider, state.serviceAccountsFilter);
```

Comment:
issue: Oh, there is no `state.serviceAccountsFilter` in this file. I think this state removed due to my previous comment.. 😂